hardware/arduino/avr/libraries/HID/HID.cpp

```diff
@@ -85,19 +85,19 @@ void HID_::AppendDescriptor(HIDDescriptorListNode *node)
 	sizeof_hidReportDescriptor += (uint16_t)node->length;
 }
 
-void HID_::SendReport(u8 id, const void* data, int len)
+void HID_::SendReport(uint8_t id, const void* data, const uint16_t len)
```

Comment:
This should be int or you should change all underlying APIs as well. I know that this is sometimes good or bad but in this case we should keep the int. 32k bytes is far enough anyways.